runtime/src/test/java/org/eclipse/xpanse/runtime/WebhookApiTest.java

```diff
@@ -0,0 +1,355 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.bind.annotation.CrossOrigin;
+
+
+/**
+ * test for WebhookApi.
+ */
+@Slf4j
+@ExtendWith(SpringExtension.class)
+@CrossOrigin
+@SpringBootTest(properties = {"spring.profiles.active=zitadel,zitadel-testbed,terraform-boot"})
+@AutoConfigureMockMvc
+public class WebhookApiTest {
+
+    private static final String userId = "adminId";
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    private static ServiceTemplateVo serviceTemplateVo;
+    private static ServiceDetailVo serviceDetailVo;
+    private static Ocl ocl;
+    private static String taskId = "dc26fbe6-0d82-4dea-8d6a-13b357c9dbd1";
+    private static String id;
+    private static ServiceDeploymentState state;
+    @Resource
+    private DeployService deployService;
+
+    @Resource
+    private MockMvc mockMvc;
+
+
+    @BeforeAll
+    static void configureObjectMapper() {
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.registerModule(new SimpleModule().addSerializer(OffsetDateTime.class,
+                OffsetDateTimeSerializer.INSTANCE));
+        objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+        objectMapper.configure(DeserializationFeature.ADJUST_DATES_TO_CONTEXT_TIME_ZONE, false);
+    }
+
+    @Test
+    @WithMockJwtAuth(authorities = RoleConstants.ROLE_ADMIN,
+            claims = @OpenIdClaims(sub = "adminId", preferredUsername = "adminName"))
+    void testDeployCallbackSuccess() throws Exception {
+        testDeployCallback();
+        boolean deployCallbackSuccess = deployCallbackSuccess(UUID.fromString(taskId));
+    }
+
+    @Test
+    @WithMockJwtAuth(authorities = RoleConstants.ROLE_ADMIN,
+            claims = @OpenIdClaims(sub = "adminId", preferredUsername = "adminName"))
+    void testDeployCallbackSuccessThrowsException() throws Exception {
+        deployCallbackThrowsException();
+        testGetServiceDetailsThrowsException();
+    }
+
+    @Test
+    @WithMockJwtAuth(authorities = RoleConstants.ROLE_ADMIN,
+            claims = @OpenIdClaims(sub = "adminId", preferredUsername = "adminName"))
+    void testDestroyCallbackSuccess() throws Exception {
+        testDestroyCallback();
+        boolean destroyCallbackSuccess = destroyCallbackSuccess(UUID.fromString(taskId));
+    }
+
+    @Test
+    @WithMockJwtAuth(authorities = RoleConstants.ROLE_ADMIN,
```

Comment:
acctually the WebHookApi does not have any authentication. So we do not need these annotations and no need to test for all roles.